Packages/vcs/Lib/vcsvtk/boxfillpipeline.py

```diff
@@ -0,0 +1,383 @@
+                                self._scalarRange[1])
+            if len(levs2) == 1: # constant value ?
+                levs2 = [levs2[0], levs2[0] + .00001]
+            self._contourLevels = []
+            if vcs2vtk.isTrue(self._gm.ext_1):
+                ## user wants arrow at the end
+                levs2[0] = -1.e20
+            if vcs2vtk.isTrue(self._gm.ext_2):
+                ## user wants arrow at the end
+                levs2[-1] = 1.e20
+            for i in range(len(levs2) - 1):
+                self._contourLevels.append([levs2[i], levs2[i+1]])
+        else:
+            if not isinstance(self._gm.levels[0], (list, tuple)):
+                self._contourLevels = []
+                levs2 = self._gm.levels
+                if numpy.allclose(levs2[0], 1.e20):
+                    levs2[0] = 0
+                for i in range(len(levs2) - 1):
+                    self._contourLevels.append([levs2[i], levs2[i+1]])
+
+        # Contour colors:
+        self._contourColors = self._gm.fillareacolors
+        if self._contourColors is None:
+            # TODO BUG levs2 may not be defined here...
+            self._contourColors = vcs.getcolors(levs2, split=0)
+
+    def _createPolyDataFilter(self):
+        """Overrides baseclass implementation."""
+        self._vtkPolyDataFilter = vtk.vtkDataSetSurfaceFilter()
+        if self._useCellScalars:
+            p2c = vtk.vtkPointDataToCellData()
+            p2c.SetInputData(self._vtkDataSet)
+            self._vtkPolyDataFilter.SetInputConnection(p2c.GetOutputPort())
+        else:
+            self._vtkPolyDataFilter.SetInputData(self._vtkDataSet)
+
+    def _plotInternal(self):
+        """Overrides baseclass implementation."""
+        # Special case for custom boxfills:
+        if self._gm.boxfill_type != "custom":
+            self._plotInternalBoxfill()
+        else:
+            self._plotInternalCustomBoxfill()
+
+        if self._maskedDataMapper is not None:
+            self._mappers.insert(0, self._maskedDataMapper)
+
+        x1, x2, y1, y2 = vcs.utils.getworldcoordinates(self._gm,
+                                                       self._data1.getAxis(-1),
+                                                       self._data1.getAxis(-2))
+
+        # And now we need actors to actually render this thing
+        actors = []
+        for mapper in self._mappers:
+            act = vtk.vtkActor()
+
+            # TODO Charles, why would mapper be a list? The custom boxfill code
+            # pushes single-element lists into the mapper list, though a comment
+            # suggests that this means a textured pattern should be used. I'll
+            # clean this up to be more clear, but need to understand the current
+            # usage pattern first.
+            if isinstance(mapper, list):
+                act.SetMapper(mapper[0])
+            else:
+                mapper.Update()
+                act.SetMapper(mapper)
```

Comment:
@doutriaux1 This was a common question I had when reading this code. The `mappers` variable would have a mix of `vtkMapper` objects and single-element lists containing a `vtkMapper`, and I can't figure out the difference or why this would be. Comments for these sort of non-obvious conventions help immensely ;)

So, what is the difference between `mapper` and `[mapper,]` in the `mappers` list?